### ISSUE TITLE
TfidfVectorizer IllegalArgumentException issue while converting to PMML

### ISSUE THREAD

**Issue by hazai** [jpmml/jpmml-sklearn#28]
I'm trying to convert a pipeline saved as pkl into a pmml file using the command: 
java -jar converter-executable-1.2-SNAPSHOT.jar --pkl-input pipeline.pkl.z --pmml-output pipeline.pmml

And I get this:
```
SEVERE: Failed to convert
java.lang.IllegalArgumentException: l2
        at sklearn.feature_extraction.text.TfidfVectorizer.encodeFeatures(TfidfVectorizer.java:45)
        at sklearn_pandas.DataFrameMapper.encodeFeatures(DataFrameMapper.java:71)
        at sklearn2pmml.PMMLPipeline.encodePMML(PMMLPipeline.java:102)
        at org.jpmml.sklearn.Main.run(Main.java:133)
        at org.jpmml.sklearn.Main.main(Main.java:99)

Exception in thread "main" java.lang.IllegalArgumentException: l2
        at sklearn.feature_extraction.text.TfidfVectorizer.encodeFeatures(TfidfVectorizer.java:45)
        at sklearn_pandas.DataFrameMapper.encodeFeatures(DataFrameMapper.java:71)
        at sklearn2pmml.PMMLPipeline.encodePMML(PMMLPipeline.java:102)
        at org.jpmml.sklearn.Main.run(Main.java:133)
        at org.jpmml.sklearn.Main.main(Main.java:99)
```

Same thing happens, of course, when I try the directly from sklearn2pmml:
```
sklearn2pmml(pipeline, "pipeline.pmml", with_repr=True)
```

Versions:
sklearn 0.18.1
joblib 0.10.3
sklearn-pandas 1.3.0
sklearn2pmml 0.17.0

**Comment by vruusmann** [comment]
Sorry, you cannot use "l1" or "l2" normalization at the moment.

Simply set the `norm` attribute to `None`:
```
TfidfVectorizer(.., norm = None)
```

**Comment by hazai** [comment]
When do you expect having the normalization part done?

I changed the code and now I get this error:
> java.lang.IllegalArgumentException: The tokenizer object (null) is not Splitter
	at sklearn.feature_extraction.text.CountVectorizer.getTokenizer(CountVectorizer.java:214)
	at sklearn.feature_extraction.text.CountVectorizer.encodeDefineFunction(CountVectorizer.java:145)
	at sklearn.feature_extraction.text.TfidfVectorizer.encodeDefineFunction(TfidfVectorizer.java:55)
	at sklearn.feature_extraction.text.CountVectorizer.encodeFeatures(CountVectorizer.java:103)
	at sklearn.feature_extraction.text.TfidfVectorizer.encodeFeatures(TfidfVectorizer.java:48)
	at sklearn_pandas.DataFrameMapper.encodeFeatures(DataFrameMapper.java:71)
	at sklearn2pmml.PMMLPipeline.encodePMML(PMMLPipeline.java:102)
	at org.jpmml.sklearn.Main.run(Main.java:133)
	at org.jpmml.sklearn.Main.main(Main.java:99)


The definition of the vectorizer is as follows:
`TfidfVectorizer(token_pattern=r'(?u)\S+', min_df=3, max_df=0.7, sublinear_tf=True, norm=None)`

**Comment by vruusmann** [comment]
> When do you expect having the normalization part done?

Not any time soon.

> java.lang.IllegalArgumentException: The tokenizer object (null) is not Splitter
> at sklearn.feature_extraction.text.CountVectorizer.getTokenizer(CountVectorizer.java:214)

You could open CountVectorizer.java, scroll to line 214 and see yourself what's happening there.

In brief, Scikit-Learn and PMML use different tokenization approaches. If you want to have a guarantee about the reproducibility of predictions, you must use `sklearn2pmml.feature_extraction.text.Splitter` as your tokenizer implementation.

Here are my notes about converting Bag-of-Words models to PMML: http://www.slideshare.net/VilluRuusmann/bagofwords-models-in-pmml

**Comment by hazai** [comment]
Thank you @vruusmann
